website/src/components/Stats/Stats.components.tsx

```diff
@@ -38,30 +39,191 @@ type MessageTreeStateStatsProps = {
+    if (stateCol === -1) {
+      continue;
+    }
+
+    const threadCount = stat.stats[i].count;
+    rows[langRow][stateCol] = threadCount;
+  }
+
+  headers = headers.map((item) => t(getTypeSafei18nKey(item)));
+
+  // Add the language code in brackets after the language name
+  rows.forEach((row) => {
+    const langCode = row[0];
+    row[0] = getLocaleDisplayName(langCode) + " [" + langCode + "]";
+  });
+
+  return { headers, rows };
+}
+
 export const MessageTreeStateStats = ({ stat, titleFn }: MessageTreeStateStatsProps) => {
   const { t } = useTranslation("stats");
+  const color = useColorModeValue(colors.light.text, colors.dark.text);
+  const uniqueLangs = Array.from(new Set(stat.stats.map((s) => s.lang)));
+  const [selectedLang, setSelectedLang] = useState("en");
+  const stats = stat.stats.filter((item) => item.lang === selectedLang);
+
   return (
     <>
-      {stat.stats.map((item) => {
-        return (
-          <Text key={item.state}>
-            <Text as="span" fontWeight="semibold">
-              {titleFn ? titleFn(item.lang) : item.lang}
-            </Text>
-            : {item.count} ({t(getTypeSafei18nKey(item.state))})
-          </Text>
-        );
-      })}
+      <Select cursor="pointer" mb={3} value={selectedLang} onChange={(e) => setSelectedLang(e.target.value)}>
+        {uniqueLangs.map((lang) => (
+          <option key={lang} value={lang}>
+            {titleFn(lang)}
+          </option>
+        ))}
+      </Select>
+      <Box minH={330}>
+        <Doughnut
+          width={100}
+          height={50}
+          options={getDefaultChartOptions(color)}
+          data={{
+            labels: stats.map((item) => t(getTypeSafei18nKey(item.state))),
+            datasets: [
+              {
+                data: stats.map((item) => item.count),
+              },
+            ],
+          }}
+        />
+      </Box>
     </>
   );
 };
 
-export const statComponents: Record<StatNames, React.FC<{ stat: StatType }>> = {
-  human_messages_by_lang: ({ stat }: KeyPairStatProps) => <KeyPairStat stat={stat} titleFn={getLocaleDisplayName} />,
-  human_messages_by_role: ({ stat }: KeyPairStatProps) => <KeyPairStat stat={stat} />,
-  message_trees_by_state: ({ stat }: KeyPairStatProps) => <KeyPairStat stat={stat} />,
+export const MessageTreeStateStatsStacked = ({ stat, titleFn }: MessageTreeStateStatsProps) => {
+  const { t } = useTranslation("stats");
+  const uniqueLangs = useMemo(() => Array.from(new Set(stat.stats.map((s) => s.lang))), stat.stats);
+  const uniqueStates = useMemo(() => Array.from(new Set(stat.stats.map((s) => s.state))), stat.stats);
+
+  const bar_options = {
+    responsive: true,
+    scales: {
+      x: {
+        stacked: true,
+      },
+      y: {
+        stacked: true,
+      },
+    },
+  };
+
+  const bar_datasets = [];
+  uniqueStates.forEach((state) => {
+    const data_state = stat.stats.filter((item) => item.state === state);
+
+    const lang_count = {};
```

Comment:
Please could we use camelCase throughout? For example `barDatasets`

Reply:
Absolutely! Sorry, lately I work in Python and I make this kind of mistakes.